### ISSUE TITLE
refactor all random functions to applyreducs

### ISSUE THREAD

**Issue by asinghvi17** [JuliaGeo/GeometryOps.jl#176]

**Comment by rafaqz** [comment]
E.g. flatten, unwrap, reconstruct

**Comment by asinghvi17** [comment]
I think only `flatten` can be done this way - here's how:

```julia
function _flatten(f, target, geoms)
	return applyreduce(f, vcat, target, geoms)
end
```
(I think)

**Comment by rafaqz** [comment]
I think even apply itself can just use applyreduce! - or they can all share a more generic/abstract function if we think that is nicer.

I like the idea of having just one GeoInterface walker

**Comment by asinghvi17** [comment]
Yeah agreed, we should do that.  What's the best approach though?  I had considered having an `Apply` struct, which carries the decompose, operate and reconstruct functions as type parameters.  This could then have pretty flexible dispatch.

The other alternative is to have an apply iterator, but that would be hard to thread.
